servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -1131,7 +1131,39 @@ void TextureStorage::_texture_2d_update(RID p_texture, const Ref<Image> &p_image
 void TextureStorage::texture_2d_update(RID p_texture, const Ref<Image> &p_image, int p_layer) {
 	_texture_2d_update(p_texture, p_image, p_layer, false);
 }
+void TextureStorage::texture_2d_update_partial(RID p_texture, const Ref<Image> &p_src_image, Rect2i p_region, Vector2i p_dst, int p_mipmap, int p_layer) {
+	ERR_FAIL_COND(p_src_image.is_null() || p_src_image->is_empty());
+	ERR_FAIL_COND_MSG(p_src_image->is_compressed(), "Compressed image is not supported");
+
+	Texture *texture = texture_owner.get_or_null(p_texture);
+	Image::Format src_format = p_src_image->get_format();
+	ERR_FAIL_COND(!texture);
+	ERR_FAIL_COND(texture->is_render_target);
+	ERR_FAIL_COND(p_dst.x < 0 || p_dst.y < 0);
+	ERR_FAIL_COND(p_region.position.x < 0 || p_region.position.y < 0 || p_region.position.x + p_region.size.x > p_src_image->get_width() || p_region.position.y + p_region.size.y > p_src_image->get_height());
+	ERR_FAIL_COND(src_format != texture->format);
+	ERR_FAIL_COND(p_mipmap < 0 || p_mipmap >= texture->mipmaps);
+
+	if (texture->type == TextureStorage::TYPE_LAYERED) {
+		ERR_FAIL_INDEX(p_layer, texture->layers);
+	}
+#ifdef TOOLS_ENABLED
+	texture->image_cache_2d.unref();
+#endif
+	Ref<Image> validated = p_src_image;
+	if (src_format == Image::FORMAT_RGB8) {
+		validated = p_src_image->get_region(p_region);
+		validated->convert(Image::FORMAT_RGBA8);
```

Comment:
Why are you converting to an alpha format? If it is to enforce 4-byte alignment, see how ``_texture_update()`` handles it. You should be able to copy that approach directly. 

We should really avoid doing a full texture convert in such cases.